crates/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -89,7 +89,7 @@ impl StringFunction {
         use StringFunction::*;
         match self {
             #[cfg(feature = "concat_str")]
-            ConcatVertical(_) | ConcatHorizontal(_) => mapper.with_same_dtype(),
+            ConcatVertical(_) | ConcatHorizontal(_) => mapper.with_dtype(DataType::Utf8),
```

Comment:
I'm not 100% certain that this change is correct as I found that we specifically changed the dtype from `mapper.with_dtype(DataType::Utf8)` to ` mapper.with_same_dtype()` in #9457.

@ritchie46 Do you still remember the purpose of making that change. 🤔

Reply:
Yes this is correct. I didn't think the input types would be non-string.